File: examples/src/main/java/de/maibornwolff/domainchoreograph/examples/statistik/orchestration/StatistikService.java
package de.maibornwolff.domainchoreograph.examples.statistik.orchestration;

import de.maibornwolff.domainchoreograph.examples.statistik.domaintypes.*;

import java.util.ArrayList;

public class StatistikService {

    public static GesamtAuswertung calculateGesamtAuswertung(StichprobenPool datenpool) {

        ArrayList<Statistik> statistikList = new ArrayList<>();

        for (Stichprobe stichprobe : datenpool.getStichprobenList()) {

            Anzahl anzahl = Anzahl.resolve(stichprobe);
            Summe summe = Summe.resolve(stichprobe);
            Mittelwert mittelwert = Mittelwert.Companion.resolve(anzahl, summe);
            Varianz varianz = Varianz.resolve(mittelwert, stichprobe, anzahl);
            StdDev stdDev = StdDev.resolve(varianz);
            Integer minimum = MinimumFactory.minimum(stichprobe);
            Statistik statistik = Statistik.resolve(stichprobe, anzahl, summe, mittelwert, varianz, stdDev, minimum);

            statistikList.add(statistik);
        }

        GesamtAuswertung gesamtAuswertung = new GesamtAuswertung(statistikList);

        return gesamtAuswertung;
    }

}

File: core/src/test/java/de/maibornwolff/domainchoreograph/example/orchestration/StatistikService.kt
package de.maibornwolff.domainchoreograph.example.orchestration

import de.maibornwolff.domainchoreograph.example.domaindefinitions.GesamtAuswertung
import de.maibornwolff.domainchoreograph.example.domaindefinitions.StichprobenService
import de.maibornwolff.domainchoreograph.example.domaindefinitions.statistik.*
import kotlin.streams.toList

object StatistikService {

    fun calculateGesamtAuswertung(stichprobenService: StichprobenService): GesamtAuswertung {

        val stichproben = stichprobenService.stichproben

        val statistikList = stichproben.stream()
            .map { stichprobe ->

                val anzahl = Anzahl.resolve(stichprobe)
                val summe = Summe.resolve(stichprobe)
                val mittelwert = Mittelwert.resolve(anzahl, summe)
                val varianz = Varianz.resolve(mittelwert, stichprobe, anzahl)
                val stdDev = StdDev.resolve(varianz)
                val minimum = MinimumDefinition.minimum(stichprobe)

                Statistik.resolve(stichprobe, anzahl, summe, mittelwert, varianz, stdDev, minimum)
            }
            .toList()

        return GesamtAuswertung(statistikList)
    }
}
